File: technical-fundamentals/coding/problems/__tests__/strings/09_stringRotation.test.ts
import stringRotation from "../../09_stringRotation";
import {reassignIsSubstring} from "../../__utils__/strings";

describe("09 - stringRotation", () => {
  beforeEach(() => {
    reassignIsSubstring()
  })

  test("rotates a string", () => {
    const str1 = "Hello";
    const str2 = "oHell";
    const result = stringRotation(str1, str2);
    expect(result).toEqual(true);
  });

  test("rotates another string", () => {
    const str1 = "waterbottle";
    const str2 = "erbottlewat";
    const result = stringRotation(str1, str2);
    expect(result).toEqual(true);
  });
});

File: technical-fundamentals/coding/problems/__tests__/strings/09_string_rotation_test.py
import unittest
from coding.problems.python_solutions.string_rotation import string_rotation

class TestStringRotation(unittest.TestCase):
    
    def test_rotates_string(self):
        str1 = "Hello"
        str2 = "oHell"
        result = string_rotation(str1, str2)
        self.assertEqual(result, True)

    def test_rotates_another_string(self):
        str1 = "waterbottle"
        str2 = "erbottlewat"
        result = string_rotation(str1, str2)
        self.assertEqual(result, True)

if __name__ == '__main__':
    unittest.main()
